code_generation/code_generation.py

```diff
@@ -106,8 +106,9 @@ def write(self):
         log.debug("Writing code subset {}".format(self.name))
         log.debug("folder: {}, file_name: {}".format(self.folder, self.file_name))
         # write the header file if it does not exist or is different
-        with open(self.headerfile + ".new", "w") as f:
-            f.write(f"ROOT::RDF::RNode {self.name}(ROOT::RDF::RNode df);")
+        f.write(
```

Comment:
As Codacy is saying, `f` is not defined anymore. You still need this line before, right?
```python
with open(self.headerfile + ".new", "w") as f:
```